sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -87,8 +89,32 @@ public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName
             UserId = userId;
             Signature = signature;
             Origin = origin;
-            States = states;
-            Headers = headers;
+            States = states ?? new Dictionary<string, object>();
```

Comment:
also update the comment to explain that the object is JsonElement and encourage user to use `GetState<T>`?